integration_test/Bindings/Python/dialects/rtl.py

```diff
@@ -5,12 +5,23 @@
 from circt.dialects import rtl
 
 from mlir.ir import *
+from mlir.dialects import builtin
 
 with Context() as ctx, Location.unknown():
     circt.register_dialects(ctx)
+
     i32 = IntegerType.get_signless(32)
-    a50 = IntegerAttr.get(i32, 50)
-    op = rtl.ConstantOp(i32, a50)
 
-    # CHECK: %{{.+}} = rtl.constant 50 : i32
-    op.print()
+    m = builtin.ModuleOp()
+    with InsertionPoint.at_block_begin(m.body):
+        op = rtl.RTLModuleOp(
```

Comment:
If you want to go full-pythonista, you can add more sugar ala `FuncOp.from_py_func`: https://github.com/llvm/llvm-project/blob/3fd64cc7a361ae02f4c0f84f5f6a85a6f05c100c/mlir/lib/Bindings/Python/mlir/dialects/_builtin_ops_ext.py#L101


Example: 
* https://github.com/llvm/llvm-project/blob/3fd64cc7a361ae02f4c0f84f5f6a85a6f05c100c/mlir/test/Bindings/Python/dialects/builtin.py#L30
* https://github.com/llvm/llvm-project/blob/3fd64cc7a361ae02f4c0f84f5f6a85a6f05c100c/mlir/test/Bindings/Python/dialects/linalg/opdsl/emit_structured_generic.py#L57

(ie. a lot of times some decorator sugar around a `def` reads/scopes a lot nicer versus inline lambdas)

Reply:
Nice! That's exactly what I wanted to do in the future. Would you be open to a patch which pulls this out and generalizes it so that all function like ops could use it?

Reply:
I missed this, that is great.

> Would you be open to a patch which pulls this out and generalizes it so that all function like ops could use it?

I think one of the best ways CIRCT pushes on the upstream infrastructure is by generalizing things from FuncOp to FunctionLike, since FunctionLike things are pervasive in hardware, but none of them are really a FuncOp. I've done some of this before, and I'm happy to help out here.

One thing I've found is "all FunctionLike ops" usually means "all FunctionLike ops that use FunctionType for the signature", which includes everything in CIRCT and most of the upstream ones, but it's worth noting the difference.

Reply:
I did a little copy-paste to try this out for now, and made it build CIRCT ops instead. It looks pretty sweet. I also added a bit to pass the argument name attributes in from the Python function's parameters.

It would be great to unify these.. not much is different.

Reply:
That's really nice. It shouldn't be hard to make the upstream version more general so that FunctionLike ops inherit -- just need to fiddle with it and get a feel for how best to structure it. We could move most of the upstream FuncOp to a new FunctionLikeOpView in _ods_common and then have tablegen choose the different base class based on traits.

Reply:
I will play around with the idea of FunctionLikeOpView